qoala/lang/hostlang.py

```diff
@@ -115,8 +116,8 @@ def __eq__(self, other: object) -> bool:
     @classmethod
     def from_generic_args(
         cls,
-        result: Optional[str],
-        args: Union[List[str], List[IqoalaTuple]],
+        result: Optional[IqoalaVar],
+        args: Union[List[IqoalaVar]],
```

Comment:
Since the parser passes an empty list `[]` for `args` instead of `None` I believe just removing `Union` is fine. There is no need to add `Optinal` here.